File: 22_08/22_08_09/pro-lifeboat.py
def solution(people, limit):
    people.sort()
    small, big = 0, len(people)-1
    cnt = 0
    while small <= big:
        cnt+=1
        if people[small] + people[big] <= limit:
            small +=1
        big -=1
    return cnt
    
print(solution([70, 80, 50],100))

File: 22_09/22_09_05/pro-life_boat.js
function solution(people, limit) {
    people.sort((a,b) => a-b)
    let small = 0
    let big = people.length-1;
    cnt = 0
    while (small <= big){
        if (people[small] + people[big] <=limit){
            small++;
        }
        big--;
        cnt++;
    }
    return cnt
}

console.log(solution([70, 50, 80, 50], 100))